### PR TITLE
linux: update to 5.19.y

### PR DESCRIPTION
**Pull request by heitbaum**
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19.1
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19.2
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19.3
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19.4
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19.5
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19.6
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19.7 

5.19.1 - 21 patches (PBRSB Mitigation)
5.19.2 - 1157 patches (Hantro)
5.19.3 - 7 patches
5.19.4 - 362 patches
5.19.5 - 1 patch
5.19.6 - 158 patches
5.19.7 - 72 patches

errors / fixes / issues / regressions
- more speculation fixes https://www.phoronix.com/news/Linux-PBRSB-Mitigation

### log 
- https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/?h=linux-5.19.y
- https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable-rc.git/log/?h=linux-5.19.y
- https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/log/
- https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable-rc.git/log/?h=queue/5.19

### 5.19.7-rc1 Build tested on all of:

```
PROJECT=Allwinner ARCH=arm DEVICE=A64 s/build linux
PROJECT=Allwinner ARCH=arm DEVICE=H3 s/build linux
PROJECT=Allwinner ARCH=arm DEVICE=H5 s/build linux
PROJECT=Allwinner ARCH=arm DEVICE=H6 s/build linux
PROJECT=Rockchip ARCH=arm DEVICE=RK3288 s/build linux
PROJECT=Rockchip ARCH=arm DEVICE=RK3328 s/build linux
PROJECT=Rockchip ARCH=arm DEVICE=RK3399 s/build linux
PROJECT=NXP ARCH=arm DEVICE=iMX6 s/build linux
PROJECT=NXP ARCH=arm DEVICE=iMX8 s/build linux
PROJECT=Qualcomm ARCH=arm DEVICE=Dragonboard s/build linux
PROJECT=Generic ARCH=x86_64 DEVICE=Generic s/build linux
PROJECT=Generic ARCH=x86_64 DEVICE=Generic-legacy s/build linux
PROJECT=Samsung ARCH=arm DEVICE=Exynos s/build linux`

### 5.19.y Run tested on all of:
- Allwinner all - tested - TBA - jernejsk
- Allwinner H6 (Tanix TX6) - 5.19.1-rc1 - heitbaum
- Generic Generic (Intel TGL - NUC11) - 5.19.7-rc1 - heitbaum
- NXP iMX6 (Cubox-i4Pro) - TBA - heitbaum
- Rockchip RK3399pro (Rock Pi N10) - TBA - heitbaum
- RK all - tested - TBA - knaerzche
- Samsung Exynos (Hardkernel ODROID XU4) - 5.19.7-rc1 - heitbaum

### CODE REVIEW

**Comment by heitbaum** [comment]
```
  CC [M]  drivers/staging/media/rkvdec/rkvdec-hevc.o
drivers/staging/media/rkvdec/rkvdec-hevc.c: In function 'assemble_hw_rps':
drivers/staging/media/rkvdec/rkvdec-hevc.c:2217:59: error: 'const struct v4l2_hevc_dpb_entry' has no member named 'pic_order_cnt'; did you mean 'pic_order_cnt_val'?
 2217 |                         if (dpb[sl_params->ref_idx_l0[i]].pic_order_cnt[0] > sl_params->slice_pic_order_cnt)
      |                                                           ^~~~~~~~~~~~~
      |                                                           pic_order_cnt_val
drivers/staging/media/rkvdec/rkvdec-hevc.c:2227:59: error: 'const struct v4l2_hevc_dpb_entry' has no member named 'pic_order_cnt'; did you mean 'pic_order_cnt_val'?
 2227 |                         if (dpb[sl_params->ref_idx_l1[i]].pic_order_cnt[0] > sl_params->slice_pic_order_cnt)
      |                                                           ^~~~~~~~~~~~~
      |                                                           pic_order_cnt_val
In file included from drivers/staging/media/rkvdec/rkvdec-hevc.c:15:
drivers/staging/media/rkvdec/rkvdec-hevc.c: In function 'config_registers':
drivers/staging/media/rkvdec/rkvdec-hevc.c:2390:91: error: 'const struct v4l2_hevc_dpb_entry' has no member named 'pic_order_cnt'; did you mean 'pic_order_cnt_val'?
 2390 |                 reg = RKVDEC_POC_REFER(i < decode_params->num_active_dpb_entries ? dpb[i].pic_order_cnt[0] : 0);
      |                                                                                           ^~~~~~~~~~~~~
drivers/staging/media/rkvdec/rkvdec-regs.h:152:59: note: in definition of macro 'RKVDEC_POC_REFER'
  152 | #define RKVDEC_POC_REFER(x)                             ((x) & 0xffffffff)
      |                                                           ^
make[4]: *** [scripts/Makefile.build:249: drivers/staging/media/rkvdec/rkvdec-hevc.o] Error 1
```

File: projects/Rockchip/devices/RK3328/linux/default/linux.aarch64.conf [hunk 1]
```diff
@@ -6245,7 +6247,6 @@ CONFIG_CRYPTO_CRC32C=y
 CONFIG_CRYPTO_CRC32=y
 CONFIG_CRYPTO_XXHASH=m
 CONFIG_CRYPTO_BLAKE2B=m
-CONFIG_CRYPTO_BLAKE2S=m
```
Comment (diederikdehaas) [inline projects/Rockchip/devices/RK3328/linux/default/linux.aarch64.conf:6248]: I don't know if it's useful, but there is a ``CONFIG_CRYPTO_BLAKE2S_ARM`` module now, which was recently [enabled on Debian](https://salsa.debian.org/kernel-team/linux/-/commit/2388a96aa63c1f06d1630544da37c8cfd1b484fc)

Comment (heitbaum) [reply projects/Rockchip/devices/RK3328/linux/default/linux.aarch64.conf:6248]: Hi @diederikdehaas _GENERIC is used on arm64, as there is no arch specific BLAKE2 for arm64 (there is for x86 and arm)
- https://github.com/torvalds/linux/tree/master/arch/arm64/crypto
- https://github.com/torvalds/linux/tree/master/arch/arm/crypto
- https://github.com/torvalds/linux/tree/master/arch/x86/crypto

the Debian reference will be for 32-bit kernels.

```
$ git grep CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S
projects/Allwinner/linux/linux.arm.conf:CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S=y
projects/Generic/linux/linux.x86_64.conf:CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S=y
projects/NXP/devices/iMX6/linux/linux.arm.conf:CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S=y
projects/RPi/devices/RPi/linux/linux.arm.conf:CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S=y
projects/RPi/devices/RPi2/linux/linux.arm.conf:CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S=y
projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf:CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S=y
projects/Samsung/linux/linux.arm.conf:CONFIG_CRYPTO_ARCH_HAVE_LIB_BLAKE2S=y
```

Comment (diederikdehaas) [reply projects/Rockchip/devices/RK3328/linux/default/linux.aarch64.conf:6248]: Oh, you're entirely right :+1: Thanks for clearing that up :-)

**Comment by jernejsk** [comment]
ffmpeg branch updated

**Comment by heitbaum** [comment]
ffmpeg patch added and merge block removed.

**Review by knaerzche — APPROVED** [review]

**Pull request merged**